packages/react-api/src/hooks/useCartoLayerProps.js

```diff
@@ -46,18 +47,22 @@ export default function useCartoLayerProps({
     };
   }
 
+  const { filterRange, filterValueUpdateTriggers, getFilterValue } = getGpuFilter(
+    source?.filters
+  );
+
   return {
     ...props,
     uniqueIdProperty,
     data: source?.data,
     type: source?.type,
     connection: source?.connection,
     credentials: source?.credentials,
-    getFilterValue: _buildFeatureFilter({ filters: source?.filters, type: 'number' }),
-    filterRange: [1, 1],
-    extensions: [new DataFilterExtension({ filterSize: 1 })],
+    getFilterValue: getFilterValue,
+    filterRange: filterRange,
+    extensions: [new DataFilterExtension({ filterSize: MAX_GPU_FILTERS })],
     updateTriggers: {
-      getFilterValue: source?.filters
+      getFilterValue: filterValueUpdateTriggers
     }
```

Comment:
It's great to see how all pieces fit together, and we'll improve perf a lot, without changing external api. Really nice job here

Reply:
Agree! Great job 😄